# Bioinformatics Methodologies to Explore Cancerous Comorbidities of Obesity

#### Abstract: 

Obesity is a complex, multi-factorial disease that’s strongly related to multiple comorbidities. Several types of cancer, diabetes mellitus, cardiovascular disease, disability, osteoarthritis, hypertension, and stroke contain comorbidities with obesity. Nowadays, the association between obesity and various types of cancer has received much attention. Bioinformatics approaches, such as gene set enrichment analysis and semantic similarity, can help evaluate the interconnection between obesity and different types of cancer. In this work, a detailed survey of gene expression data on obesity and various cancers was compiled from the NCBI public repository. Then, a pipeline is designed that integrates gene expression, cell type data, and online resources using bioinformatics methods such as gene set enrichment analysis and semantic similarity. Using this pipeline, we examined the differentially expressed genes, gene ontology terms, and pathways associated with obesity and various types of cancers. Therefore, the closeness among various pathologies indicates the disease ontology terms.

## Authors
- [@Md. Shahriare Satu](https://scholar.google.com/citations?user=iBFrdREAAAAJ&hl=en)
- [@Shuvo Saha Roy](https://www.github.com/ShuvoSahaRoy)
- [@Koushik Howlader](https://scholar.google.com/citations?hl=en&user=eLvPVrcAAAAJ)
- [@Md. Habibur Rahman]()
- [@Mohammad Ali Moni](https://scholar.google.com/citations?hl=en&user=JnwcUwkAAAAJ)



## Installation

Please install **R**, **RStudio**, and all required dependencies. The specific package versions used in this study are listed in the manuscript.

This repository contains two primary R scripts:

1. **Script 1** – Gene enrichment analysis  
2. **Script 2** – Semantic similarity analysis  

For additional reference, you may consult the provided `.xlsx` files or visit the following source for more details:  
[Main source of the original code](http://bioinformatica.isa.cnr.it/COELIAC_DISEASE/SCRIPTS/)

---

## Usage

You may run **Script 1** directly for your own dataset.  
However, to **reproduce the results of this study**, please proceed as follows:

1. Execute all scripts located inside each **dataset folder**.  
   - Each dataset folder contains code specific to that dataset.  
   - Running these scripts will generate a `.txt` file.

2. Use the generated `.txt` file(s) as input for **Script 2**.


## License

[MIT](https://choosealicense.com/licenses/mit/)

    

    
